src/actions/proxy.spec.ts

```diff
@@ -3,204 +3,216 @@ import { createTinlake, TestProvider } from '../test/utils'
-    //   const issueTx = await borrowerTinlake.proxyTransferIssue(proxyAddr, testConfig.nftRegistry, tokenId)
-    //   const issueResult = await borrowerTinlake.getTransactionReceipt(issueTx)
-
-    //   // console.log('issueResult', issueResult)
-
-    //   assert.equal(issueResult.status, SUCCESS_STATUS)
-    //   assert.equal(await borrowerTinlake.getNFTOwner(testConfig.nftRegistry, tokenId), proxyAddr)
-
-    //   // set loan parameters and fund tranche
-    //   const loanId = await borrowerTinlake.nftLookup(testConfig.nftRegistry, tokenId)
-    //   const amount = '10'
-    //   const relyTx = await governanceTinlake.relyAddress(adminAccount.address, contractAddresses['NFT_FEED'])
-    //   const relyResult = await governanceTinlake.getTransactionReceipt(relyTx)
-
-    //   // console.log('relyResult', relyResult)
-
-    //   const nftfeedId = await adminTinlake.getNftFeedId(testConfig.nftRegistry, Number(tokenId))
-    //   const updateNftTx = await adminTinlake.updateNftFeed(nftfeedId, Number(amount))
-    //   const updateNftResult = await adminTinlake.getTransactionReceipt(updateNftTx)
+  describe('borrow - lend - repay - redeem flow', async () => {
+    it('success: full loan cycle', async () => {
 
-    //   // console.log('updateNftResult', updateNftResult)
+      await governanceTinlake.setMinimumJuniorRatio('0')
+      const ratio = await governanceTinlake.getMaxJuniorRatio()
+      assert.equal(ratio, 1000000000000000000000000000)
 
-    //   await fundTranche('100')
-    //   const initialTrancheBalance = await borrowerTinlake.getCurrencyBalance(contractAddresses['JUNIOR_TRANCHE'])
+      await governanceTinlake.setMinimumEpochTime('1')
+      const epoch = await governanceTinlake.getMinimumEpochTime()
+      assert.equal(epoch.toString(), '1')
 
-    //   // borrow
-    //   const borrowTx = await borrowerTinlake.proxyLockBorrowWithdraw(proxyAddr, loanId, amount, borrowerAccount.address)
-    //   // console.log('borrowTx', borrowTx)
+      await governanceTinlake.setMinimumChallengeTime('1')
+      const challenge = await governanceTinlake.getChallengeTime()
+      assert.equal(challenge.toString(), '1')
 
-    //   const borrowResult = await borrowerTinlake.getTransactionReceipt(borrowTx)
+      // add governance address to memberlists
+      await governanceTinlake.updateJuniorMemberList(contractAddresses['GOVERNANCE'], validityTimestamp)
+      await governanceTinlake.updateSeniorMemberList(contractAddresses['GOVERNANCE'], validityTimestamp)
 
-    //   // console.log('borrowResult', borrowResult)
+      // create new proxy and mint collateral NFT to borrower
+      const proxyAddr = await borrowerTinlake.proxyCreateNew(borrowerAccount.address)
+      const tokenId = await governanceTinlake.mintTitleNFT(testConfig.nftRegistry, borrowerAccount.address)
+      const approveTx = await borrowerTinlake.approveNFT(testConfig.nftRegistry, tokenId, proxyAddr)
+      await verifyTransactionSuccess(approveTx)
```

Comment:
Great idea to add this helper! Definitely makes the intent of the tests much clearer.